### PR TITLE
Remove duplications from `Explorer.Series`

### PR DESCRIPTION
**Pull request by philss**
This makes some functions lean and add helpers to check for types.

### CODE REVIEW

**Review by cigrainger — APPROVED** [review]

**Comment by cigrainger** [comment]
I like it!

**Comment by philss** [comment]
@cigrainger I was thinking after open the PR: maybe I should use multiple function clauses instead of `case` (with the `{left, right}` tuple) . Do you prefer that?

**Comment by cigrainger** [comment]
Oh -- yes. I think multiple function clauses is better. I often find myself doing exactly that with case or cond statements on tuples.

**Comment by philss** [comment]
@cigrainger updated. I think it looks better now :smiley:

**Comment by philss** [comment]
I'm merging this. If you think something could be improved, let me know so I can fix :)

**Comment by cigrainger** [comment]
Looks good to me!

File: lib/explorer/series.ex [hunk 1]
```diff
@@ -38,6 +38,10 @@ defmodule Explorer.Series do
 
   @compile {:no_warn_undefined, Nx}
 
+  defguardp numeric_dtype?(dtype) when dtype in [:float, :integer]
+  defguardp numeric_or_bool_dtype?(dtype) when dtype in [:float, :integer, :boolean]
+  defguardp numeric_or_date_dtype?(dtype) when dtype in [:float, :integer, :date, :datetime]
```
Comment (josevalim) [inline lib/explorer/series.ex:43]: The convention for guards is to use the `is_` prefix with no question mark. :)

**Pull request merged**